### ISSUE TITLE
MySQL 8: ordering of multiple table options can prevent diff

### ISSUE THREAD

**Issue by trevorr** [skeema/skeema#88]
After migrating from local MySQL 5.6.45 to RDS MySQL 8.0.15, `skeema diff` starting reporting this error on 16 out of 73 tables. I ran `skeema init` with no errors, and then ran `skeema diff`:

```
2019-07-25 23:56:09 [WARN]  Skipping table `categories`: unable to generate DDL due to use of unsupported features. Use --debug for more information.
2019-07-25 23:56:09 [DEBUG] --- Expected CREATE
2019-07-25 23:56:09 [DEBUG] +++ MySQL-actual SHOW CREATE
2019-07-25 23:56:09 [DEBUG] @@ -15 +15 @@
2019-07-25 23:56:09 [DEBUG] -) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci ROW_FORMAT=DYNAMIC CHECKSUM=1 DELAY_KEY_WRITE=1
2019-07-25 23:56:09 [DEBUG] +) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci CHECKSUM=1 DELAY_KEY_WRITE=1 ROW_FORMAT=DYNAMIC
```

The only difference seems to be in the ordering. I built skeema from source a day or two ago. The same schema had no problems when it was running on local MySQL 5.6.

```
$ skeema --version
skeema version 1.2.5-dev (snapshot build from source)
```

**Comment by evanelias** [comment]
Thank you for the issue report! I've confirmed this is a bug in Skeema affecting MySQL 8.0. I should have a fix in master within a few days. 

If you need an urgent work-around, you could manually remove the `CHECKSUM=1` and `DELAY_KEY_WRITE=1` clauses from your tables if you are using InnoDB storage engine. These clauses have no effect whatsoever on InnoDB; they're only meaningful to MyISAM tables. (see https://dev.mysql.com/doc/refman/8.0/en/create-table.html#create-table-options for corresponding manual entries for `CHECKSUM` and `DELAY_KEY_WRITE`.)

Background on this Skeema bug:

* In all previous versions of MySQL, the order of *table option clauses* in `SHOW CREATE TABLE` exactly matched what was shown in `information_schema.tables.create_options`.

* MySQL 8 introduced a new internal *data dictionary*, which has changed the underlying implementation of much of `information_schema`. As a result, the ordering of several things there has changed and no longer matches the behavior of previous versions.

* Skeema primarily uses `information_schema` to introspect table structure (rather than relying on SQL parsing, which could be brittle across different versions/vendors). In order to sanity-check its understanding of a table and ability to diff a table, Skeema ensures that its expectation of what the canonical `CREATE TABLE` looks like matches the output of `SHOW CREATE TABLE`. The bug fix will be to ignore differences in the order of table options in MySQL 8+, as long as the option names and values are otherwise identical.

**Comment by trevorr** [comment]
Thanks! Manually removing those unnecessary table options worked.

**Comment by evanelias** [comment]
Glad the work-around was successful!

I've fixed the root cause in the master branch now too. Next binary release might be a few weeks away though.

Thanks again for the issue report!

**Comment by evanelias** [comment]
I've just released [Skeema v1.2.6](https://github.com/skeema/skeema/releases/tag/v1.2.6) containing this fix. Thanks again for the bug report!
